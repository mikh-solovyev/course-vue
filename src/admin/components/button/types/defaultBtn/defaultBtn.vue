<template>
  <button
      :class="['default-btn-container', 'btn-decorator', {disabled}, {plain}]"
      v-if="typeAttr !== 'file'"
      :type="typeAttr"
      v-on="$listeners"
      :disabled="disabled"
  >{{title}}</button>

  <label class="btn-file-container" v-else-if="typeAttr === 'file'">
    <div :class="['btn-file-fake', 'btn-decorator', {notGround}]">{{title}}</div>
    <input class="btn-file-input" type="file" v-on="$listeners" />
  </label>
</template>
<script>
export default {
  props: {
    title: {
      type: String,
      default: "Отправить"
    },
    disabled: {
      type: Boolean,
      default: false
    },
    plain: Boolean,
    notGround: Boolean,
    typeAttr: {
      type: String,
      default: "button",
      validator: value => ["button", "file", "submit"].includes(value)
    }
  }
};
</script>

<style lang="postcss" scoped src="./defaultBtn.pcss"></style>