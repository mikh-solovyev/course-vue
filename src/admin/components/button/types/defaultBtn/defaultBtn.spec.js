import defaultBtn from "./defaultBtn.vue";
import { mount } from "@vue/test-utils";

it("component render", async () => {
    const wrapper = mount(defaultBtn);
    const component = await wrapper.find(".btn-decorator");

    expect(component.exists()).toBe(true);
});

it("component render type file", async () => {
    const wrapper = mount(defaultBtn);

    await wrapper.setProps({
        typeAttr: "file"
    });

    const fileElement = await wrapper.find(".btn-file-input");
    expect(fileElement.exists()).toBe(true);
});

it("component class added", async () => {
    const wrapper = mount(defaultBtn);

    await wrapper.setProps({
        disabled: true
    });

    const fileElement = await wrapper.find(".btn-decorator");
    expect(fileElement.classes("disabled")).toBe(true);
});

it("component btn container text", async () => {
    const wrapper = mount(defaultBtn);

    await wrapper.setProps({
        title: "Text"
    });

    const fileElement = await wrapper.find(".btn-decorator");
    expect(fileElement.text()).toEqual("Text");
});

it("snapshots", async () => {
    const wrapper = mount(defaultBtn);

    expect(wrapper.element).toMatchSnapshot();
});