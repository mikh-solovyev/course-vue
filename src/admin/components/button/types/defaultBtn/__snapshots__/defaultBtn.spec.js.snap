// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<button
  class="default-btn-container btn-decorator"
  type="button"
>
  Отправить
</button>
`;
